### PR TITLE
Fixes to get gl backend to work with opengl 3.0

### PR DESCRIPTION
**Pull request by ZeGentzy**
Signed-off-by: Hal Gentz <zegentzy@protonmail.com>

Fixes #issue
PR checklist:
- [ ] `make` succeeds (on *nix)
- [ ] `make reftests` succeeds
- [x] tested examples with the following backends: Linux, quad
- [ ] `rustfmt` run on changed code

### CODE REVIEW

File: src/backend/gl/src/device.rs [hunk 1]
```diff
@@ -1071,7 +1090,28 @@ impl d::Device<B> for Device {
...
+                        let mut h = h;
+                        for i in 0..num_levels {
+                            gl.TexImage2D(
+                                gl::TEXTURE_2D,
+                                i as _,
+                                int_format as _,
+                                w as _,
+                                h as _,
+                                0,
+                                gl::RGBA,
```
Comment (msiglreith) [inline src/backend/gl/src/device.rs]: hardcoded values

File: src/backend/gl/src/queue.rs [hunk 2]
```diff
@@ -564,6 +564,20 @@ impl CommandQueue {
                 let gl = &self.share.context;
                 gl.BindSampler(index, sampler);
             }
+            com::Command::SetTextureSamplerSettings(index, texture, ref sinfo) => unsafe {
+                let gl = &self.share.context;
+                gl.ActiveTexture(gl::TEXTURE0 + index);
+                gl.BindTexture(gl::TEXTURE_2D, texture);
```
Comment (msiglreith) [inline src/backend/gl/src/queue.rs]: we should at least warn/panic here is the texture is not 2D

Comment (ZeGentzy) [reply src/backend/gl/src/queue.rs]: Currently the OGL backend does not keep track of dimensionality, as `create_image` currently only supports making 2D images, therefore, all textures will be 2D. All the other code in the backend also assumes textures will only be 2D. Solving this problem is beyond the scope of this PR.

File: src/backend/gl/src/info.rs [hunk 3]
```diff
@@ -304,6 +305,7 @@ impl Info {
 pub(crate) fn query_all(gl: &GlContainer) -> (Info, Features, LegacyFeatures, Limits, PrivateCaps) {
     use self::Requirement::*;
     let info = Info::get(gl);
+    println!("{:?}", info);
```
Comment (kvark) [inline src/backend/gl/src/info.rs]: debug leftover?

File: src/backend/gl/src/device.rs [hunk 4]
```diff
@@ -410,6 +410,58 @@ impl Device {
...
+pub(crate) unsafe fn set_sampler_info<SetParamFloat, SetParamFloatVec, SetParamInt>(
+    share: &Starc<Share>,
+    info: &i::SamplerInfo,
+    mut set_param_float: SetParamFloat,
+    mut set_param_float_vec: SetParamFloatVec,
+    mut set_param_int: SetParamInt,
+)
+where
+    SetParamFloat: FnMut(GLenum, GLfloat),
+    SetParamFloatVec: FnMut(GLenum, &GLfloat),
```
Comment (kvark) [inline src/backend/gl/src/device.rs]: this should be accepting `&[GLfloat]`

File: src/backend/gl/src/info.rs (hunk 3 above)
Comment (ZeGentzy) [reply src/backend/gl/src/info.rs]: Yup, removed.

**Pull request merged**